File: dotnet/Liberator.Driver/Performance/RatTimer.cs
using System;

namespace Liberator.Driver.Performance
{
    /// <summary>
    /// Timer for RatDriver
    /// </summary>
    public class RatTimer : ITimer
    {
        /// <summary>
        /// Start time for the timer
        /// </summary>
        public DateTime StartTime { get; set; }

        /// <summary>
        /// End time for the timer
        /// </summary>
        public DateTime EndTime { get; set; }

        /// <summary>
        /// Duration for the timer instance
        /// </summary>
        public TimeSpan Duration { get; set; }

        /// <summary>
        /// Constructor for the RatTimer class
        /// </summary>
        public RatTimer()
        {
            StartTime = DateTime.Now;
            EndTime = DateTime.Now;
            Duration = new TimeSpan(0, 0, 0, 0, 0);
        }

        /// <summary>
        /// Starts the time and resets the duration
        /// </summary>
        public void Start()
        {
            StartTime = DateTime.Now;
            EndTime = DateTime.Now;
            Duration = new TimeSpan(0, 0, 0, 0, 0);
        }

        /// <summary>
        /// Stops the timer and calculates the duration
        /// </summary>
        public void Stop()
        {
            EndTime = DateTime.Now;
            Duration = EndTime - StartTime;
        }
    }
}

File: java/src/main/java/org/liberator/ratdriver/performance/RatTimer.java
package org.liberator.ratdriver.performance;

import java.time.Instant;

public class RatTimer {

    RatTimer(){
        startTime = Instant.now();
        endTime = Instant.now();
        duration = -1;
    }

    public Instant startTime;
    public Instant endTime;
    public long duration;

    /**
     * Start the timer
     */
    public void Start() {
        startTime = Instant.now();
        endTime = Instant.now();
        duration = 0;
    }

    /**
     * Stop the timer
     */
    public void Stop() {
        endTime = Instant.now();
        duration = endTime.toEpochMilli() - startTime.toEpochMilli();
    }
}
